x/checkers/keeper/msg_server_create_game.go

```diff
@@ -2,16 +2,34 @@ package keeper
 
 import (
 	"context"
+	"strconv"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	rules "github.com/xavierlepretre/checkers/x/checkers/rules"
 	"github.com/xavierlepretre/checkers/x/checkers/types"
 )
 
 func (k msgServer) CreateGame(goCtx context.Context, msg *types.MsgCreateGame) (*types.MsgCreateGameResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// TODO: Handling the message
-	_ = ctx
+	nextGame, found := k.Keeper.GetNextGame(ctx)
+	if !found {
+		panic("NextGame not found")
+	}
+	newIndex := strconv.FormatUint(nextGame.IdValue, 10)
+	newGame := types.FullGame{
```

Comment:
why not create a storedgame right away? it helps avoids unnecessary conversions

Reply:
In future branches, my `FullGame` assists with conversions, for instance to `Coin` or with dates comparison. Do you suggest that I instead _add_ such helper functions to the `StoredGame` type?
I actually thought about it at some point, never did the jump, though.